Src/ILGPU/Backends/IL/ILEmitterExtensions.cs

```diff
@@ -67,10 +67,21 @@ public static class ILEmitterExtensions
                 emitter.Emit(LocalOperation.Load, result);
 
                 emitter.Emit(OpCodes.Ldarg_0);
-                emitter.Emit(OpCodes.Ldflda, field);
-                emitter.EmitCall(field.FieldType.GetMethod(
+
+                var fieldHashCode = field.FieldType.GetMethod(
                     GetHashCodeInfo.Name,
-                    BindingFlags.Public | BindingFlags.Instance));
+                    BindingFlags.Public | BindingFlags.Instance);
+                if (field.FieldType.IsValueType &&
+                    fieldHashCode.DeclaringType != field.FieldType)
+                {
+                    emitter.Emit(OpCodes.Ldfld, field);
+                    emitter.Emit(OpCodes.Box, field.FieldType);
+                } else
+                {
```

Comment:
MINOR: Coding convention.
```suggestion
                }
                else
                {
```

Reply:
This should be fixed now.